packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3512,7 +3518,17 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
       if (!widget.readOnly) {
         _scheduleShowCaretOnScreen(withAnimation: true);
       }
-      if (!_value.selection.isValid) {
+      if (kIsWeb && !_isMultiline && !_nextFocusChangeIsInternal) {
```

Comment:
To make the code more self-explanatory, would it help to put these into a well-named boolean, e.g:
```suggestion
      final bool shouldSelectAll = kIsWeb && !_isMultiline && !_nextFocusChangeIsInternal;
      if (shouldSelectAll) {
```